### PR TITLE
Add an about dialog showing licenses.

### LINKED ISSUE #172: [FEATURE] Need to have an about dialog that displays version and dependent licenses.
Because we use other open source software that needs to have their licenses displayed, and because it might be important to know version information when troubleshooting, we need to have an "About" dialog with appropriate information in it.

### PR DESCRIPTION
**Pull request by gspencergoog**
This adds an about dialog so that we can show all of the open source licenses and application information.

@joshua-s, please make sure the information I included is correct, like the copyright notice for the app, etc.

Fixes https://github.com/coronavirus-diary/coronavirus-diary/issues/172

### CODE REVIEW

File: lib/src/ui/screens/home/home.dart [hunk 1]
```diff
@@ -190,3 +192,42 @@ class _HomeScreenState extends State<HomeScreen> {
     );
   }
 }
+
+class _MainMenu extends StatelessWidget {
+  static const String copyright = '© 2020 Josh Smith';
```
Comment (ferndot) [inline lib/src/ui/screens/home/home.dart]: Perhaps, "An open source project from the Coronavirus Diary community" or "Developed by the Coronavirus Diary community under the MIT license"?

Comment (gspencergoog) [reply lib/src/ui/screens/home/home.dart]: Would you like a link to the webpage and/or github repo here?

Comment (ferndot) [reply lib/src/ui/screens/home/home.dart]: That would be great!

Comment (gspencergoog) [reply lib/src/ui/screens/home/home.dart]: OK, how's this look? 

Screenshot_1585865410

Comment (ferndot) [reply lib/src/ui/screens/home/home.dart]: That looks great, I think you could condense it down into:
```
This app is developed as an open source project by the Coronavirus Diary community. To learn more and view the source code, please visit <link>Coronavirus Diary on GitHub</link>.
```

Please add the strings to l10n.

Comment (gspencergoog) [reply lib/src/ui/screens/home/home.dart]: OK, I updated the string to be similar to what you wrote, but realized that when internationalizing it, I can't embed a link in the internationalized text (limitation of the Intl package).  I reworded it slightly and put the link below it.

Screenshot_1585876998

Comment (ferndot) [reply lib/src/ui/screens/home/home.dart]: Perfect!

File: lib/src/l10n/app_en.arb [hunk 2]
```diff
@@ -1,6 +1,37 @@
...
+    "description": "Description in the 'About' box, above a hyperlink to the source website."
+  },
+
+  "aboutBoxLinkText": "Coronavirus Diary on GitHub",
+    "@aboutBoxLinkText":  {
+      "description": "Text label on a hyperlink to the source code for the app."
+    },
+
+
+  "aboutBoxCopyright": "Copyright © 2020 Josh Smith",
```
Comment (ferndot) [inline lib/src/l10n/app_en.arb]: This is unused and can be removed

Comment (gspencergoog) [reply lib/src/l10n/app_en.arb]: No, that one is used, on [line 189](https://github.com/coronavirus-diary/coronavirus-diary/pull/174/files#diff-1cace224e2fa2d64dbc63e90682b4c46R189) of home.dart.

**Review by ferndot — APPROVED** [review]

**Pull request merged**